src/cloud_provider/service.rs

```diff
@@ -533,6 +537,22 @@ pub fn scale_down_application(
     )
 }
 
+pub fn delete_router<T>(target: &DeploymentTarget, service: &T, is_error: bool) -> Result<(), EngineError>
+where
+    T: Router,
+{
+    send_progress_on_long_task(service, crate::cloud_provider::service::Action::Delete, || {
+        if service.has_custom_domains() {
+            let (kubernetes, environment) = match *target {
+                DeploymentTarget::ManagedServices(k, env) => (k, env),
+                DeploymentTarget::SelfHosted(k, env) => (k, env),
+            };
+            do_stateless_service_cleanup(kubernetes, environment, &service.custom_router_helm_release_name())?
```

Comment:
can you please rename it and avoid stuffs starting with do_, it can be confusing with other methods. Thx

Reply:
done